pkg/build/nodeimage/buildcontext.go

```diff
@@ -308,19 +290,28 @@ func (c *buildContext) prePullImages(bits kube.Bits, dir, containerID string) ([
 		}
 	}()
 
-	// create a plan of image loading
-	loadFns := []func() error{}
-	for _, image := range pulled {
-		image := image // capture loop var
-		loadFns = append(loadFns, func() error {
-			f, err := os.Open(image)
-			if err != nil {
-				return err
+	fns := []func() error{}
+	for _, image := range requiredImages {
+		image := image // https://golang.org/doc/faq#closures_and_goroutines
+		fns = append(fns, func() error {
+			if !builtImages.Has(image) {
+				err := importer.Pull(image, dockerBuildOsAndArch(c.arch))
+				if err != nil {
+					c.logger.Warnf("Failed to pull %s with error: %v", image, err)
+					runE := exec.RunErrorForError(err)
+					c.logger.Warn(string(runE.Output))
+				}
+				// TODO(bentheelder): generate a friendlier name
 			}
-			defer f.Close()
-			return importer.LoadCommand().SetStdout(os.Stdout).SetStderr(os.Stdout).SetStdin(f).Run()
+			return nil
 		})
 	}
+	if err := errors.AggregateConcurrent(fns); err != nil {
```

Comment:
if never returns an error :shrug:

Reply:
this line cannot know that, `AggregateConcurrent` can return it's own error for example.

Reply:
That is true